nknd.go

```diff
@@ -375,3 +384,112 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func GetChordID(seeds []string, publickey []byte) ([]byte, error) {
+	id, err := chain.DefaultLedger.Store.GetID(publickey)
+	if err == nil {
+		if len(id) != 0 && !bytes.Equal(id, make([]byte, 32)) {
+			return id, nil
+		}
+	}
+
+	log.Warningf("get chord ID from localnode faild, no chord ID in ledger")
+
+	rand.Shuffle(len(seeds), func(i int, j int) {
+		seeds[i], seeds[j] = seeds[j], seeds[i]
+	})
+
+	for _, seed := range seeds {
+		id, err := client.GetChordID(seed, publickey)
+		if err == nil {
+			return id, nil
```

Comment:
we should also check length here just in case the seed node is broken or malicious